### PR TITLE
feat: move ce permissions ctas and more unit tests

### PR DESCRIPTION
**Pull request by berzerkeer**
## Description

Introduce tests for context menu and hardcoded permissions for CE.

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested

- Manual
- Jest

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### CODE REVIEW

File: app/client/src/pages/Editor/DataSourceEditor/Connected.tsx [hunk 1]
```diff
@@ -49,10 +46,10 @@ function Connected() {
   );
   const pagePermissions = useSelector(getPagePermissions);
 
-  const canCreateDatasourceActions = isPermitted(
-    [...userWorkspacePermissions, ...pagePermissions],
-    [PERMISSION_TYPE.CREATE_DATASOURCE_ACTIONS, PERMISSION_TYPE.CREATE_ACTIONS],
-  );
+  const canCreateDatasourceActions = hasCreateDatasourceActionPermission([
+    ...userWorkspacePermissions,
```
Comment (berzerkeer) [inline app/client/src/pages/Editor/DataSourceEditor/Connected.tsx:50]: @ankitakinger should I make that change here ? That this should be `datasourcePermissions` and not `userWorkspacePermission`

Comment (ankitakinger) [reply app/client/src/pages/Editor/DataSourceEditor/Connected.tsx:50]: Thanks for ointing it out I'd made this change by mistake on EE. Have pushed it to CE.

File: app/client/src/pages/Editor/Explorer/EntityExplorer.test.tsx [hunk 2]
```diff
@@ -276,4 +308,25 @@ describe("Entity Explorer tests", () => {
...
+  //   store.dispatch({
+  //     type: ReduxActionTypes.FETCH_ACTIONS_SUCCESS,
+  //     payload: mockApiDatas,
+  //   });
+  //   store.dispatch({
+  //     type: ReduxActionTypes.FETCH_JS_ACTIONS_SUCCESS,
+  //     payload: mockJsActions,
+  //   });
+  //   jest
+  //     .spyOn(entitiesSelectors, "selectFilesForExplorer")
```
Comment (ankitakinger) [inline app/client/src/pages/Editor/Explorer/EntityExplorer.test.tsx:321]: should we remove these comments? or this is intentionally kept? @berzerkeer

File: app/client/src/components/editorComponents/GlobalSearch/GlobalSearchHooks.tsx [hunk 3]
```diff
@@ -92,33 +98,43 @@ export const useFilteredFileOperations = (query = "") => {
...
-              action: (pageId: string, from: EventLocation) =>
-                createNewQueryAction(pageId, from, ds.id),
-            }))
-          : []),
+        ...filteredAppWideDS.map((ds) =>
+          hasCreateDatasourceActionPermission([
+            ...(ds.userPermissions ?? []),
+            ...pagePermissions,
+          ])
+            ? {
```
Comment (ankitakinger) [inline app/client/src/components/editorComponents/GlobalSearch/GlobalSearchHooks.tsx:106]: what if this returns a blank array? I think you should add array length check here as well as at other places in this file.

**Review by ankitakinger — APPROVED** [review]
LGTM

**Pull request merged**